libs/langchain/langchain/document_loaders/csv_loader.py

```diff
@@ -46,17 +48,21 @@ def __init__(
             csv_args: A dictionary of arguments to pass to the csv.DictReader.
               Optional. Defaults to None.
             encoding: The encoding of the CSV file. Optional. Defaults to None.
+            autodetect_encoding: Whether to try to autodetect the file encoding.
         """
         self.file_path = file_path
         self.source_column = source_column
         self.encoding = encoding
         self.csv_args = csv_args or {}
+        self.autodetect_encoding = autodetect_encoding
 
     def load(self) -> List[Document]:
         """Load data into document objects."""
 
         docs = []
-        with open(self.file_path, newline="", encoding=self.encoding) as csvfile:
+        if self.autodetect_encoding:
+            detected_encodings = detect_file_encodings(self.file_path)
+        with open(self.file_path, newline="", encoding=detected_encodings if self.autodetect_encoding else self.encoding) as csvfile:
```

Comment:
This has different behavior than `TextLoader` in that even if an encoding is defined, the autodetected one would by mistake take precedence

Reply:
I changed the code.
Could you please review it?